x/ccv/provider/keeper/grpc_query.go

```diff
@@ -105,39 +107,34 @@ func (k Keeper) GetConsumerChain(ctx sdk.Context, consumerId string) (types.Chai
 	}, nil
 }
 
-// TODO (PERMISSIONLESS)
 func (k Keeper) QueryConsumerChainStarts(goCtx context.Context, req *types.QueryConsumerChainStartProposalsRequest) (*types.QueryConsumerChainStartProposalsResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "This query is not supported anymore. Use `QueryConsumersThatAreAboutToStart` instead")
+}
+
+func (k Keeper) QueryConsumersThatAreAboutToStart(goCtx context.Context, req *types.QueryConsumersThatAreAboutToStartRequest) (*types.QueryConsumersThatAreAboutToStartResponse, error) {
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	var props []*types.ConsumerAdditionProposal
 
-	for _, prop := range k.GetAllPendingConsumerAdditionProps(ctx) {
-		// prevent implicit memory aliasing
-		p := prop
-		props = append(props, &p)
-	}
-
-	return &types.QueryConsumerChainStartProposalsResponse{Proposals: &types.ConsumerAdditionProposals{Pending: props}}, nil
+	consumerIds := k.GetInitializedConsumersReadyToLaunch(ctx, math.MaxUint32)
+	return &types.QueryConsumersThatAreAboutToStartResponse{ConsumerIds: consumerIds}, nil
 }
 
-// TODO (PERMISSIONLESS)
 func (k Keeper) QueryConsumerChainStops(goCtx context.Context, req *types.QueryConsumerChainStopProposalsRequest) (*types.QueryConsumerChainStopProposalsResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "This query is not supported anymore. Use `QueryConsumersThatAreAboutToStop` instead")
+}
+
+func (k Keeper) QueryConsumersThatAreAboutToStop(goCtx context.Context, req *types.QueryConsumersThatAreAboutToStopRequest) (*types.QueryConsumersThatAreAboutToStopResponse, error) {
```

Comment:
ditto: This query will return consumers that will stop in the next `BeginBlock`. Not sure I understand the use case for it.

Reply:
Removed these queries altogether.